File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/TeacherResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TeacherResponse {
    private String id;

    private String username;

    private String email;

    private String firstName;

    private String lastName;

    private String dob;

    private String teacherId;

    private String departmentId;

    private String educationalUnitId;

    private String description;

    private String socialUrl;

    private String bankAccountNumber;

    private DepartmentResponse department;

    private EducationalUnitResponse educationalUnit;

    private String accountStatus;

    private String avatarUrl;

    private String phoneNumber;

    private String bio;
}

File: react-type-vite/src/services/api/response/teacherResponse.ts
import type { DepartmentResponse } from "./departmentResponse";
import type { EducationalUnitResponse } from "./educationalUnitResponse";

export interface TeacherResponse {
  id: string;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  dob?: string;
  teacherId: string;
  departmentId?: string;
  educationalUnitId?: string;
  description?: string;
  avatarUrl?: string;
  bankAccountNumber?: string;
  department?: DepartmentResponse;
  educationalUnit?: EducationalUnitResponse;
  accountStatus: string
}